thewalrus/csamples.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Classical Sampling Algorithms
+=============================
+
+.. currentmodule:: thewalrus.csamples
+
+This submodule provides access to classical sampling algorithms.
+
+Summary
+-------
+.. autosummary::
+    rescale_adjacency_matrix_thermal
+    rescale_adjacency_matrix
+    generate_thermal_samples
+
+Code details
+------------
+"""
+# pylint: disable=too-many-arguments
+
+import numpy as np
+from scipy.optimize import root_scalar
+
+def rescale_adjacency_matrix_thermal(
+    A, n_mean, check_positivity=True, check_symmetry=True, rtol=1e-05, atol=1e-08
+):
+    r""" Returns the scaling parameter by which the adjacency matrix A
+    should be rescaled so that the Gaussian state that encodes it has
+    a total mean photon number n_mean for thermal sampling.
+    Args:
+        A (array): Adjacency matrix, assumed to be positive semi-definite and real
+        n_mean (float): Mean photon number of the Gaussian state
+        check_positivity (bool): Checks if the matrix A is positive semidefinite
```

Comment:
What happens if `check_positivity = True` and we feed in a non psd A?

Reply:
i.e. should we say that `check_positivity = True` causes an exception to be raised if we don't get a psd?